base_layer/core/src/blockchain/chain.rs

```diff
@@ -55,18 +53,44 @@ pub struct Chain {
 }
 
 impl Chain {
-    pub fn new() -> Result<Chain, ChainError> {
-        let chain = Chain {
-            block_chain_state: BlockchainState::new()?,
-            orphans: HashMap::new(),
-            current_total_pow: ProofOfWork::default(),
-        };
-        Ok(chain)
+
+    /// Adds a new block to the persistent state.
+    ///
+    /// This process includes:
+    ///   * Adds the range proof hashes to the range proof MMR
+    ///   * Marks all inputs in the block as spent
+    ///   * Adds all outputs to the UTXO set and MMR
+    ///   * Adds the transaction kernels to the kernel MMR
+    ///   * Adds the block header to the header MMR
+    /// An error is returned if:
+    ///   * the block has already been added (by checking whether the block header hash exists already)
+    ///   * any of the inputs were not in the UTXO set or were marked as spent already
+    ///   * There was a problem creating the checkpoints
+    pub fn add_new_block(&mut self, new_block: &Block) -> Result<(), StateError> {
```

Comment:
It might be better to move the Block in here so that the caller knows that clones are happening and they can choose to clone it themselves before calling it

Reply:
Yeah sure